openedx/core/djangoapps/credit/signals.py

```diff
@@ -70,3 +82,209 @@ def listen_for_grade_calculation(sender, username, grade_summary, course_key, de
+
+    """
+    # TODO: Refactor code (move code to proper files e.g., 'common/lib/xmodule/xmodule/partitions/partitions.py') in once ICRV access control functionality is complete
+
+    # TODO: Dynamically update user_id (actual course publisher)
+    user_id = 1
+
+    # TODO: Increase logging (add logging for errors and success)
+    # get user partition with provided partition scheme
+    user_partition = UserPartition.get_scheme(partition_scheme)
+    if user_partition is None:
+        # log and raise exception 'NoSuchUserPartitionError' if no matching
+        # user partition exists
+        log.error(
+            u'No user partition found with scheme "%s". Exiting tagging course content with user partition.',
+            partition_scheme
+        )
+        raise NoSuchUserPartitionError
+
+    access_control_credit_xblocks = _get_credit_xblocks_for_access_control(course_key)
+    new_user_partitions = []
+    for block in access_control_credit_xblocks:
+        # for now we are entertaining only verification partition scheme
+        if partition_scheme == VERIFICATION_SCHEME:
+            # create group configuration for VerificationPartitionScheme
+            group_configuration = _verification_partition_group_configuration(user_partition, block)
+            new_user_partitions.append(group_configuration)
+
+            # update the 'group_access' of provided xblock 'block' and its ancestor.
+            _update_content_group_access(user_id, block, group_configuration)
+
+    # Now add all newly created partition in 'user_partitions' field of course
+    _update_course_user_partitions(course_key, user_id, new_user_partitions)
+
+
+def _access_dict(group_configuration, access_groups_id_list):
+    """ Returns dict 'group_access' of provided user partition and its groups.
+
+    Args:
+        group_configuration  (UserPartition): UserPartition object
+        access_groups_id_list  (List): List of id's of partition groups
+
+    Returns:
+        Dict object.
+
+    """
+    access_dict = {
+        group_configuration.id: [
+            group_configuration.get_group(group_id).id for group_id in access_groups_id_list
+        ]
+    }
+    return access_dict
+
+
+def _update_content_group_access(user_id, block, group_configuration):
+    """ Update the 'group_access' of provided xblock 'block' and its ancestor.
+
+    Args:
+        user_id (Int): ID of user who has publish the course
+        block (xblock): XBlock mixin
+        group_configuration  (UserPartition): UserPartition object
+
+    """
+    # update 'group_access' field of gating xblock with groups
+    # 'verified_allow' and 'verified_deny' of provided user partition
+    # 'group_configuration'
+    access_groups_id_list = [VerificationPartitionScheme.VERIFIED_ALLOW, VerificationPartitionScheme.VERIFIED_DENY]
+    block.group_access = _access_dict(group_configuration, access_groups_id_list)
+    modulestore().update_item(block, user_id)
```

Comment:
@wedaly @clintonb me and @awais786 were thinking of one other possibility that is, using the same user id which last edited the module.

1. Lets consider the scenario that there are 2 different verticals/units [`vertical 1`, `vertical 2` (with an ICRV)]. `vertical 1` edited by a `user A` and `vertical 2` edited by another `user B`. This editing information can be seen on each vertical/unit e.g., `Last published Aug 07, 2015 at 07:38 UTC by user-1`.
2. After some time `user A` edits the `vertical 1` and publish it, this will trigger our method `tag_course_content_with_partition_scheme` to create user partitions and update access groups of gated block. 

**Approach 1: Somehow we are able to get publishing user in signal and add it while updating a block**
On publish since we update ICRV (present in `vertical 2`) `group_access` field and save it (while tagging course content) so after publish not only `vertical 1` but `vertical 2` both will start displaying `user A` as the last editor e.g., `Last published Aug 07, 2015 at 07:56 UTC by user-A`, although `user A` didn't directly edited/updated `vertical 2`.
**Approach 2: Use last user which updated that block (from block's `edit_info`) and add it while updating a block**
Now `vertical 1` and `vertical 2` will keep displaying `user A` and `user B`as their respective editors (although the time of update will still change which is due to using hard-code current time `now()` on update) (**this is somewhat consistent with current behavior**).